scripts/generateAssetData/utils/portals.ts

```diff
@@ -0,0 +1,143 @@
+import {
+  arbitrumChainId,
+  ASSET_NAMESPACE,
+  avalancheChainId,
+  baseChainId,
+  bscChainId,
+  type ChainId,
+  ethChainId,
+  gnosisChainId,
+  optimismChainId,
+  polygonChainId,
+  toAssetId,
+} from '@shapeshiftoss/caip'
+import type { Asset } from '@shapeshiftoss/types'
+import axios from 'axios'
+import qs from 'qs'
+import { getAddress, isAddressEqual, zeroAddress } from 'viem'
+
+import { colorMap } from '../colorMap'
+import { createThrottle } from '.'
+
+// Non-exhaustive - https://api.portals.fi/docs#/Supported/SupportedController_getSupportedTokensV2 for full docs
+type TokenInfo = {
+  key: string
+  name: string
+  decimals: number
+  symbol: string
+  address: string
+  images: string[] | undefined
+}
+
+type GetTokensResponse = {
+  totalItems: number
+  pageItems: number
+  more: boolean
+  page: number
+  tokens: TokenInfo[]
+}
+
+const CHAIN_ID_TO_PORTALS_NETWORK: Partial<Record<ChainId, string>> = {
+  [avalancheChainId]: 'avalanche',
+  [ethChainId]: 'ethereum',
+  [polygonChainId]: 'polygon',
+  [bscChainId]: 'bsc',
+  [optimismChainId]: 'optimism',
+  [arbitrumChainId]: 'arbitrum',
+  [gnosisChainId]: 'gnosis',
+  [baseChainId]: 'base',
+}
+
+export const fetchPortalsTokens = async (
+  chainId: ChainId,
+  page: number = 0,
+  accTokens: TokenInfo[] = [],
+): Promise<TokenInfo[]> => {
+  console.log(`Fetching page ${page} of Portals tokens for chainId: ${chainId}`)
+  const url = 'https://api.portals.fi/v2/tokens'
+  const PORTALS_API_KEY = process.env.REACT_APP_PORTALS_API_KEY
+
+  const { throttle, clear } = createThrottle({
+    capacity: 100, // 100 requests per minute https://build.portals.fi/billing
```

Comment:
Fyi: Updated rpm limit to 500 and request limit to 15,000,000 (up from 100 and 50,000). See dashboard for confirmation.

Reply:
well why thank you @suhailgme! 🤯 
<img width="740" alt="image" src="https://github.com/user-attachments/assets/a0d46d88-73c4-43b9-ad04-a8cfbf66430e">

Reply:
3fea3ad6c2143d4d457fa3b76c9a1bc63854c8b8